src/main/scala/TraceIO.scala

```diff
@@ -83,7 +87,7 @@ object TraceOutputTop {
     new TraceOutputTop(proto.map(t => TracedInstructionWidths(t.head)), proto.map(_.size))
 }
 
-trait CanHaveTraceIO { this: HasTiles =>
+trait CanHaveTraceIO { outer: HasTiles =>
```

Comment:
Wait, why is it outer instead of this now? You don't make this change for any of the other traits.

Reply:
Yeah this doesn't change anything it's just providing an alias for the self type. I'll put it back.

I don't remember why i changed this, but i think i started rearchitecting this class but then stopped myself half-way without completely reverting all the changes.